mage_integrations/mage_integrations/destinations/elasticsearch/target_elasticsearch/sinks.py

```diff
@@ -0,0 +1,225 @@
+    @param stream_name:
+    @param index_format:
+    @param schemas:
+    @return: str
+    """
+    today = datetime.date.today()
+    arguments = {
+        **{
+            "stream_name": stream_name,
+        },
+        **schemas,
+    }
+    environment = jinja2.Environment()
+    template = environment.from_string(index_format)
+    return template.render(**arguments).replace("_", "-")
+
+
+def build_fields(
+    stream_name: str,
+    mapping: Dict,
+    record: Dict[str, Union[str, Dict[str, str], int]],
+    logger: singer_sdk.io_base.logger,
+) -> Dict:
+    """
+    build_fields parses records for supplied mapping to be used later in index
+    templating and ecs metadata field formulation
+    @param logger:
+    @param stream_name:
+    @param mapping: dict
+    @param record:  str
+    @return: dict
+    """
+    schemas = {}
+    if stream_name in mapping:
+        logger.debug(INDEX_TEMPLATE_FIELDS, ": ", mapping[stream_name])
+        for k, v in mapping[stream_name].items():
+            match = jsonpath_ng.parse(v).find(record)
+            if len(match) == 0:
+                logger.warning(
+                    f"schema key {k} with json path {v} could not be found in record: {record}"
+                )
+                schemas[k] = v
+            else:
+                if len(match) < 1:
+                    logger.warning(
+                        f"schema key {k} with json path {v} has multiple \
+                        associated fields, may cause side effects"
+                    )
+                schemas[k] = match[0].value
+    return schemas
+
+
+class ElasticSink(BatchSink):
+    """ElasticSink target sink class."""
+
+    max_size = 1000  # Max records to write in one batch
+
+    def __init__(
+        self,
+        target: PluginBase,
+        stream_name: str,
+        schema: Dict,
+        key_properties: Optional[List[str]],
+    ):
+        super().__init__(target, stream_name, schema, key_properties)
+        self.client = self._authenticated_client()
+
+    def build_request_body_and_distinct_indices(
+        self, records: List[Dict[str, Union[str, Dict[str, str], int]]]
+    ) -> Tuple[List[Dict[Union[str, Any], Union[str, Any]]], Set[str]]:
+        """
+        build_request_body_and_distinct_indices builds the bulk request body
+        and collects all distinct indices that will be used to create indices before bulk upload.
+        @param records:
+        @return:
+        """
+        updated_records = []
+        index_mapping = {}
+        metadata_fields = {}
+        distinct_indices = set()
+        if INDEX_TEMPLATE_FIELDS in self.config:
+            index_mapping = self.config[INDEX_TEMPLATE_FIELDS]
+        if METADATA_FIELDS in self.config:
+            metadata_fields = self.config[METADATA_FIELDS]
+
+        for r in records:
+            index = template_index(
+                self.config['table'],
+                self.config[INDEX_FORMAT],
```

Comment:
The "index_format" is not in the README and not in the config.json

Reply:
`INDEX_FORMAT` was being used to set time variables to index names
This would cause a elevated number of indexes to be created at a long period, which could cause problems to users.
To fix this, the only possible configuration for the `INDEX_FORMAT`  variable was set to be Table name ( shown in line 128)

Reply:
Also, updated docs to add the new destination in 81ab599aaebddd1d79fae5ebc20ed186777a3689